lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -2085,6 +2086,89 @@ class CommandObjectSwift_RefCount : public CommandObjectRaw {
   }
 };
 
+class CommandObjectLanguageSwiftTaskBacktrace final
+    : public CommandObjectParsed {
+public:
+  CommandObjectLanguageSwiftTaskBacktrace(CommandInterpreter &interpreter)
+      : CommandObjectParsed(interpreter, "backtrace",
+                            "Show the backtrace of Swift tasks. See `thread "
+                            "backtrace` for customizing backtrace output.",
+                            "language swift task backtrace <variable-name>") {
+    AddSimpleArgumentList(eArgTypeVarName);
+  }
+
+private:
+  void DoExecute(Args &command, CommandReturnObject &result) override {
+    if (!m_exe_ctx.GetFramePtr()) {
+      result.AppendError("no active frame selected");
+      return;
+    }
+
+    if (command.empty() || command[0].ref().empty()) {
+      result.AppendError("no task variable");
+      return;
+    }
+
+    StackFrame &frame = m_exe_ctx.GetFrameRef();
+    uint32_t path_options =
+        StackFrame::eExpressionPathOptionsAllowDirectIVarAccess;
+    VariableSP var_sp;
+    Status status;
+    ValueObjectSP valobj_sp = frame.GetValueForVariableExpressionPath(
+        command[0].c_str(), eDynamicDontRunTarget, path_options, var_sp,
+        status);
+    if (!valobj_sp) {
+      result.AppendError(status.AsCString());
+      return;
+    }
+
+    addr_t task_ptr = LLDB_INVALID_ADDRESS;
+    ThreadSafeReflectionContext reflection_ctx;
+    if (ValueObjectSP task_obj_sp =
+            valobj_sp->GetChildMemberWithName("_task")) {
+      task_ptr = task_obj_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
+      if (task_ptr != LLDB_INVALID_ADDRESS)
+        if (auto *runtime = SwiftLanguageRuntime::Get(m_exe_ctx.GetProcessSP()))
+          reflection_ctx = runtime->GetReflectionContext();
+    }
+    if (task_ptr == LLDB_INVALID_ADDRESS || !reflection_ctx) {
+      result.AppendError("failed to access Task data from runtime");
+      return;
+    }
+
+    llvm::Expected<ReflectionContextInterface::AsyncTaskInfo> task_info =
+        reflection_ctx->asyncTaskInfo(task_ptr);
+    if (auto error = task_info.takeError()) {
+      result.AppendError(toString(std::move(error)));
+      return;
+    }
+
+    auto thread_task = ThreadTask::Create(
+        task_info->id, task_info->resumeAsyncContext, m_exe_ctx);
+    if (auto error = thread_task.takeError()) {
+      result.AppendError(toString(std::move(error)));
+      return;
+    }
+
+    // GetStatus prints the backtrace.
+    thread_task.get()->GetStatus(result.GetOutputStream(), 0, UINT32_MAX, 0,
+                                 false, true);
+    result.SetStatus(lldb::eReturnStatusSuccessFinishResult);
+  }
+};
+
+class CommandObjectLanguageSwiftTask final : public CommandObjectMultiword {
+public:
+  CommandObjectLanguageSwiftTask(CommandInterpreter &interpreter)
+      : CommandObjectMultiword(
+            interpreter, "task", "Commands for inspecting Swift Tasks.",
+            "language swift task <subcommand> [<subcommand-options>]") {
```

Comment:
Side note: Should we add an alias `task` = `language swift task` or (maybe less controversial) `swift` = `language swift` that is on by default?

Reply:
Related: is `swift-healthcheck` an alias for `language swift healthcheck`?

Reply:
> Should we add an alias `task` = `language swift task` or (maybe less controversial) `swift` = `language swift` that is on by default?

one of these two would be nice, I guess the question then is which one.

> Related: is `swift-healthcheck` an alias for `language swift healthcheck`?

it's not currently, but we could switch it to that.